src/new_api_v04/mutex_tree.rs

```diff
@@ -1,27 +1,41 @@
-//! Express lists of constraints as trees of constraints.
+//! Express logic relations between constraints as trees.
 //!
 //! The idea is to define the semantics of constraints by how they interact
-//! with one another.
+//! with one another. Siblings in the tree model mutually exclusive constraints,
+//! whereas a path from the root is a conjunction of constraints: the grandchildren
+//! of a child thus represent further refinements of the child's constraint.
 //!
-//! A constraint tree is a tree along with the following data and properties:
-//!  - data
-//!     - tree edges are labelled with constraints
-//!     - tree nodes may be labelled with an integer, to be interpreted
-//!       as an index with respect to a list of constraints.
-//!  - properties
-//!     - for any edge with constraint label A, the constraint labels on children
-//!       edges B must be such that B => A.
-//!     - for a node n with index i, with constraints A1, ..., Ak on the edges
-//!       of the path from n to the root, we have
-//!                           C[i] <=> A1 ∧ A2 ∧ ... ∧ Ak
-//!       where C[i] is the i-th constraint in the list of constraints associated
-//!       with the tree.
-//!     - index labels in nodes are unique.
-
-/// A trait to define the semantics of predicate enums.
+//! More concretely, a constraint tree is a tree along with the following data:
+//!   - tree edges are labelled with constraints
+//!   - tree leaves may be labelled with one or more integers, to be interpreted as
+//!     indices with respect to a list of constraints.
+//!   - an ordering of the children for any node.
+//!
+//! Its interpretation is most easily explained by viewing the tree as a state
+//! automaton, where we say that node `n` is reachable for a given input if
+//!  - `n` is the root, or
+//!  - the parent `p` of `n` is reachable, the constraint on the edge from `p`
+//!    to `n` is satisfied and none of the constraints on the edges between `p`
+//!    and a sibling of `n` that is smaller than `n` are satisfied.
```

Comment:
A more concise rewording. Feel free to ignore.
```suggestion
//!  - the parent `p` of `n` is reachable and `n` is the smallest sibling for 
//!    which the edge constraint from `p` to the sibling is satisfied.
```